releases/release-4.0.9.md

```diff
@@ -0,0 +1,179 @@
+
++ TiFlash
+
+    - Reduce the latency of replica reads
+    - Refine TiFlash's error messages
+    - Limit the memory usage of cache data when the data volume is huge
+    - Add a monitoring metric for the number of coprocessor tasks being handled
+
++ Tools
+
+    + Backup & Restore (BR)
+
+        - Disallow the ambiguous `--checksum false` argument in the command line, which does not disable checksum. Accept `--checksum=false` only. [#588](https://github.com/pingcap/br/pull/588)
+        - Support changing the PD configuration temporarily [#596](https://github.com/pingcap/br/pull/596)
+        - Support analyzing tables after restore [#622](https://github.com/pingcap/br/pull/622)
+        - Retry for the `read index not ready` and `proposal in merging mode` errors [#626](https://github.com/pingcap/br/pull/626)
+
+    + TiCDC
+
+        - Add an alert for enabling TiKV's Hibernate Region feature [#1120](https://github.com/pingcap/ticdc/pull/1120)
+        - Reduce memory usage in the schema storage [#1127](https://github.com/pingcap/ticdc/pull/1127)
+        - Add the feature of unified sorter, which accelerates replication when the data size of the incremental scan is large (experimental) [#1122](https://github.com/pingcap/ticdc/pull/1122)
+        - Support configuring the maximum message size and the maximum message batch in the TiCDC Open Protocol message (only for Kafka sink) [#1079](https://github.com/pingcap/ticdc/pull/1079)
+
+    + Dumpling
+
+        - Retry dumping data on failed chunks [#182](https://github.com/pingcap/dumpling/pull/182)
+        - Support configuring both the `-F` and `-r` arguments at the same time [#177](https://github.com/pingcap/dumpling/pull/177)
+        - Exclude system databases in `--filter` by default [#194](https://github.com/pingcap/dumpling/pull/194)
+        - Support the `--transactional-consistency` parameter and support rebuilding MySQL connections during retry [#199](https://github.com/pingcap/dumpling/pull/199)
+
+    + TiDB Lightning
+
+        - Filter out all system schemas by default [#459](https://github.com/pingcap/tidb-lightning/pull/459)
+        - Support setting a default value for the auto-random primary key for the Local-backend or Importer-backend [#457](https://github.com/pingcap/tidb-lightning/pull/457)
+        - Use range properties to make the range split more precise in Local-backend [#422](https://github.com/pingcap/tidb-lightning/pull/422)
+        - Support a human-readable format (such as "2.5 GiB") in `tikv-importer.region-split-size`, `mydumper.read-block-size`, `mydumper.batch-size`, and `mydumper.max-region-size` [#471](https://github.com/pingcap/tidb-lightning/pull/471)
+
+    + TiDB Binlog
+
+        - Exit the Drainer process with the non-zero code if the upstream PD is down or if applying DDL or DML statements to the downstream fails [#1012](https://github.com/pingcap/tidb-binlog/pull/1012)
+
+## Bug Fixes
+
++ TiDB
```

Comment:
```
- Fix a bug that delete operation does not lock unique key [#20705](https://github.com/pingcap/tidb/pull/20705)
- Fix a bug that snapshot read will hit lock cache [#21539](https://github.com/pingcap/tidb/pull/21539)
```

Reply:
the corresponding change in docs-cn is https://github.com/pingcap/docs-cn/pull/4976#discussion_r542411298